Make the handling of NONE/None internally consistent.

freakboy3742 opened the pull request:
There is a difference between a Python `None`, and a symbolic constant `NONE = 'none'`. The symbolic `NONE` was added after the initial implementation of validated properties; however, the logic around the interpretation of the `NONE` value was incomplete/inconsistent. It made a partial attempt to map the string `'none'` to a Python `None`, but this conflicted with style definitions that had an explicitly allowed value of a symbolic `NONE`.

This PR implements a (hopefully consistent?) approach to `None`/`NONE`:

* A Python `None` is no longer a valid value for a property. If you want to represent "nothing", use a symbolic `NONE`
* An initial value of `None` is interpreted as "No initial/default value". The interpretation of this value is left to the style.
* An explicit value of `None` is interpreted as "Reset to initial value".

This doesn't break anything in Toga, as Pack uses `NONE` consistently (which led to beeware/toga#1502).

Colosseum will require some updates, as the usage there predates the introduction of the symbolic `None`.

Fixes #3.

## PR Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
- [x] All new features have been tested
- [x] All new features have been documented
- [x] I have read the **CONTRIBUTING.md** file
- [x] I will abide by the code of conduct

mhsmith commented:
I think making `None` and `NONE` mean different things is asking for trouble. How about disallowing `None` entirely in the public API? If the user wants to reset an attribute to the default value, they can do that using `del`, which I see is already implemented.

If my analysis at https://github.com/beeware/toga/issues/1502 was correct, this wouldn't break anything, because passing `None` is already broken.

freakboy3742 commented:
> I think making `None` and `NONE` mean different things is asking for trouble. How about disallowing `None` entirely in the public API? If the user wants to reset an attribute to the default value, they can do that using `del`, which I see is already implemented.

Yeah - I was on the fence on this one; it's a fine line between "a convenient interpretation" and "potentially confusing". I've removed this and added a specific warning of the edge case (otherwise the error becomes `Invalid value None for property foo; Valid values are: a, b, none` which is even more confusing).

> If my analysis at [beeware/toga#1502](https://github.com/beeware/toga/issues/1502) was correct, this wouldn't break anything, because passing `None` is already broken.

Agreed.

mhsmith commented on src/travertino/declaration.py:
```diff
@@ -172,21 +170,29 @@ def __str__(self):
             return getattr(self, "_%s" % name, initial)
 
         def setter(self, value):
+            # Treat a Python None as a reset to initial value
```
```suggestion
```
